### PR TITLE
support single output in YoloX nms

### PR DESCRIPTION
**Pull request by lkdci**

### CODE REVIEW

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 1]
```diff
@@ -80,11 +80,14 @@ def __init__(
         self.with_confidence = with_confidence
 
     def forward(self, x, device: str = None):
```
Comment (BloodAxe) [inline src/super_gradients/training/models/detection_models/yolo_base.py:82]: Could you please add type hints to x to annotate what are the supported input types will be?

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 2]
```diff
@@ -80,11 +80,14 @@ def __init__(
...
+        # Use the main output features in case of multiple outputs.
+        if isinstance(x, (tuple, list)):
```
Comment (BloodAxe) [inline src/super_gradients/training/models/detection_models/yolo_base.py:84]: Mind adding a unit test to ensure that this change passes two sanity checks? 

postprocessing_callback(model.eval().forward(dummy_input))
postprocessing_callback(model.train().forward(dummy_input))

Comment (lkdci) [reply src/super_gradients/training/models/detection_models/yolo_base.py:84]: model should be one of the detection models, i.e yolox? Or just a toy model? @BloodAxe

Comment (BloodAxe) [reply src/super_gradients/training/models/detection_models/yolo_base.py:84]: YoloX. Thanks

Comment (lkdci) [reply src/super_gradients/training/models/detection_models/yolo_base.py:95]: Sorry for the late reply, added [post prediction unit test](https://github.com/Deci-AI/super-gradients/pull/813/commits/24ac09841750118dac77dd5c54ddc066764ac07b)

**Comment by lkdci** [comment]
@BloodAxe Added unit tests with yolox output and mock model outputs. I will appreciate your re-review

File: src/super_gradients/training/models/detection_models/yolo_base.py (hunk 2 above)
Comment (lkdci) [reply src/super_gradients/training/models/detection_models/yolo_base.py:95]: `postprocessing_callback(model.train().forward(dummy_input))` will fails even without this PR since in train mode we don't decode bounding box (using DetectX module), and only logits are outputted

File: src/super_gradients/training/models/detection_models/yolo_base.py (hunk 1 above)
Comment (lkdci) [reply src/super_gradients/training/models/detection_models/yolo_base.py]: added type hints [type hints](https://github.com/Deci-AI/super-gradients/pull/813/commits/dab2d28e2e57b75198d48822deebd94ca197301a)

**Comment by shaydeci** [comment]
@lkdci  @BloodAxe This PR has been open for quite a while now. It has comments that haven't been resolved yet and conflicts.
Please fix and re-review.

**Comment by lkdci** [comment]
@BloodAxe @shaydeci , conflicts resolved. As mentioned above review was addressed, I'll appreciate your re-review of the fixes

**Review by ofrimasad — APPROVED** [review]
LGTM

**Review by ofrimasad — APPROVED** [review]
LGTM
